File: exercism/clojure/triangle/src/triangle.clj
(ns triangle)

(defn is-valid? [a b c]
  (cond 
    (not-every? (fn [n] (> n 0)) [a b c] ) false
    (< (+ a b) c) false
    (< (+ a c) b) false
    (< (+ b c) a) false
    :else true 
    )
  )

(defn equilateral? [a b c]
  (and (is-valid? a b c) (= a b c))
)

(defn isosceles? [a b c]
  (and (is-valid? a b c) (>= 2 (count (set [a b c]))))
)

(defn scalene? [a b c]
  (if (is-valid? a b c) (not (isosceles? a b c)) false)
)

File: exercism/cpp/triangle/triangle.cpp
#include "triangle.h"
#include <stdexcept>
namespace triangle
{
    short kind(double side1, double side2, double side3)
    {

        if (side1 + side2 + side3 == 0 || side1 > side2 + side3 || side2 > side1 + side3 || side3 > side1 + side2)
        {
            throw std::domain_error("Illegal arguments");
        }
        if (side1 == side2 && side2 == side3)
        {
            return triangle::flavor::equilateral;
        }
        if (side1 == side2 || side2 == side3 || side1 == side3)
        {
            return triangle::flavor::isosceles;
        }
        return triangle::flavor::scalene;
    }
} // namespace triangle
